### PR TITLE
feat: template crd

### LINKED ISSUE #4292: Template CRD for Job Templates
**Is your feature request related to a problem? Please describe.**
Currently, Testkube utilizes job templates in the form of yaml files with Go Templating for dynamic parameters within Kubernetes Jobs. However, managing these templates becomes a pain if you have different tests with different scheduling requirements, and it has a lot of overhead to edit the templates in the ConfigMap.

**Describe the solution you'd like**
I propose the introduction of a new Kubernetes Custom Resource Definition (CRD) for these templates, let's call it `Template`. By representing each template as a CRD, we can leverage Kubernetes' native abilities for versioning, watching, and validating these resources. Here's a brief outline of the potential solution:

```
apiVersion: config.testkube.io/v1alpha1
kind: Template
metadata:
  name: job-template-1
spec:
  type: job|container|scraper
  body: <could be a multi-line yaml string>
```

And then we can add a `templateRef` field in the `Test` and `TestSuite` CRDs:
```
apiVersion: tests.testkube.io/v1
kind: Test
metadata:
  name: kubeshop-cypress
spec:
  templateRef: job-template-1
  repository:
    branch: main
    path: examples
    type: git
    uri: https://github.com/kubeshop/testkube-executor-cypress.git
  type: cypress/project
```

**Describe alternatives you've considered**
Testkube already has support for this but the support has a lot of overhead and is quite limited.
Current support:
* https://docs.testkube.io/articles/creating-tests/#changing-the-default-job-template-used-for-test-execution
* https://docs.testkube.io/articles/creating-tests/#changing-the-default-cronjob-template-used-for-scheduled-test-execution

**Additional context**
By moving the templates to a dedicated CRD, we not only make the system more Kubernetes-native but also provide a more organized and scalable approach to handle dynamic test templates. This would be in line with Kubernetes' extensibility principles and would simplify the operational complexity associated with template management. Moreover, integrating RBAC, validation, and versioning on these templates would be more straightforward.

### PR DESCRIPTION
**Pull request by vsukhin**
## Pull request description 

Fix #4292 

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Comment by vsukhin** [comment]
yes, I will add  more docs

File: docs/docs/cli/testkube_create_template.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+## testkube create template
+
+Create new Template
```
Comment (jfermi) [inline docs/docs/cli/testkube_create_template.md:3]:
```suggestion
Create a new Template.
```

File: docs/docs/cli/testkube_delete_template.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+## testkube delete template
+
+Delete template
+
+### Synopsis
+
+Delete template, pass template name which should be deleted
```
Comment (jfermi) [inline docs/docs/cli/testkube_delete_template.md:7]:
```suggestion
Delete a template and pass the template name to be deleted.
```

File: docs/docs/cli/testkube_get_template.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+## testkube get template
+
+Get template details
+
+### Synopsis
+
+Get template, you can change output format, to get single details pass name as first arg
```
Comment (jfermi) [inline docs/docs/cli/testkube_get_template.md:7]:
```suggestion
Get template allows you to change the output format. To get single details, pass the template name as the first argument.
```

**Review by jfermi — COMMENTED** [review]
A few editing suggestions.

**Comment by vsukhin** [comment]
@jfermi please check templates.mdx

File: docs/docs/articles/templates.mdx [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+# Templates
+
+Templates allow you to store templates for other resources used in Testkube. We support such a list of templates job | container | cronjob | scraper | pvc | webhook. To define templates in Testkube, you'll need to provide a template body (in golang template format) and a type of the template.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:6]:
```suggestion
Templates allow you to store templates for other resources used in Testkube. We support a list of templates job | container | cronjob | scraper | pvc | webhook. To define templates in Testkube, you'll need to provide a template body (in Golang template format) and a type of the template.
```

File: docs/docs/articles/templates.mdx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+## Creating a Template
+The template can be created using the API, CLI, or a Custom Resource.
+
+<Tabs>
+<TabItem value="api" label="API">
+If you prefer to use the API for creating a template, you can check API spec for templates in below doc.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:13]:
```suggestion
If you prefer to use the API for creating a template, please visit the API spec for templates in the doc below.
```

File: docs/docs/articles/templates.mdx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+![OpenAPI spec](../openapi.md)
+
+</TabItem>
+
+<TabItem value="cli" label="CLI">
+Templates can be created with Testkube CLI using the `create template` command.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:20]:
```suggestion
Templates can be created with the Testkube CLI using the `create template` command.
```

File: docs/docs/articles/templates.mdx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+kind: Webhook
+metadata:
+  name: example-webhook
+  namespace: testkube
+spec:
+  type: job
+  body: <YOUR_TEMPLATE_BODY>
+```
+
+Where <YOUR_TEMPLATE_BODY> should be replaced with kubernetes job definition in golang template format.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:45]:
```suggestion
Where <YOUR_TEMPLATE_BODY> should be replaced with the Kubernetes job definition in Golang template format.
```

File: docs/docs/articles/templates.mdx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+```sh
+kubectl apply -f template.yaml
+```
+
+</TabItem>
+
+</Tabs>
+
+### Using template
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:57]:
```suggestion
### Using Templates
```

File: docs/docs/articles/templates.mdx [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+You will need to refer a template in corresponding reference field of the resource.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:58]:
```suggestion
You will need to refer to a template in the corresponding reference field of the resource.
```

File: docs/docs/articles/templates.mdx [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+
+<Tabs>
+<TabItem value="api" label="API">
+
+Check templateReference fields in api spec, for example, Test -> executionRequest -> jobTemplateReference field
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:63]:
```suggestion
Check templateReference fields in API spec. For example, Test -> executionRequest -> jobTemplateReference field.
```

File: docs/docs/articles/templates.mdx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+![OpenAPI spec](../openapi.md)
+
+</TabItem>
+
+<TabItem value="cli" label="CLI">
+
+Templates can be created with Testkube CLI using the `create template` command.
```
Comment (jfermi) [inline docs/docs/articles/templates.mdx:70]:
```suggestion
Templates can be created with the Testkube CLI using the `create template` command.
```

**Review by jfermi — COMMENTED** [review]
Editing suggestions for templates.mdx.

**Pull request merged**